lib/net/ssh/authentication/methods/abstract.rb

```diff
@@ -31,11 +29,7 @@ def initialize(session, options = {})
             @key_manager = options[:key_manager]
             @options = options
             @prompt = options[:password_prompt]
-            @pubkey_algorithms = options[:pubkey_algorithms] \
-              || %w[rsa-sha2-256-cert-v01@openssh.com
```

Comment:
Our default pubkeys_alogirthm will be much extended, won't it? I mean in case confg doesn't define it. 


https://github.com/net-ssh/net-ssh/pull/923/files#diff-d3f3cd24513081ec6736e7516607d8de54cb329fe6319a2c3fabe4edd67b4393R51-R65

Reply:
Yy, It is because `pubkey_algorithms` is now checked for all keys (not just RSA). (See [change in lib/net/ssh/authentication/methods/publickey.rb](https://github.com/net-ssh/net-ssh/pull/923/files#diff-f39e09d93c6f4a8374520565aa7ca0bbca6ed2c614305bc1d8997b9263612461L126))

I have included all public key algorithms, I have found in net-ssh sources, in defatults, so that default behavior remains the same (all of them still allowed), as `pubkey_algorithms` now also affects keys other than RSA.  I also kept order of RSA algorithms, relative to each other (to keep behavior for RSA keys).

Reply:
RSA keys are somewhat special, because they are the only ones, where there is not 1:1 relationship between keys and algorithms. There are multiple possible algorithms for one key (since [RFC 8332](https://datatracker.ietf.org/doc/html/rfc8332)).

Variable `pubkey_algorithms` was originally internal, used to just control order of RSA algorithms (with possibility to be extended for other algorithms later, i.e. `PubkeyAcceptedAlgorithms`). As `pubkey_algorithms` was made public by: https://github.com/net-ssh/net-ssh/pull/891 , I would like to fully implement it.